integrations/kafka_connector/src/main/java/software/amazon/timestream/utility/TimestreamWriter.java

```diff
@@ -89,26 +149,55 @@ public List<RejectedRecord> writeRecords(final AWSServiceClientFactory clientFac
             final int batchSize = records.size() / TimestreamSinkConstants.DEFAULT_BATCHSIZE + 1;
             List<Record> batchRecords = null;
             for (int currentBatch = 0; currentBatch < batchSize; currentBatch ++) {
-                try {
+                if (!writeToInfluxDB && !writeToLiveAnalytics) {
+                    // no target specified, cannot write, send records to DLQ
                     batchRecords = getBatchRecords(records, currentBatch);
-                    if (batchRecords != null && !batchRecords.isEmpty()) {
-                        final WriteRecordsRequest writeRequest = WriteRecordsRequest.builder()
-                                .databaseName(databaseName)
-                                .tableName(tableName)
-                                .records(batchRecords)
-                                .build();
-                        final WriteRecordsResponse writeResponse = clientFactory.getTimestreamClient().writeRecords(writeRequest);
-                        LOGGER.debug("DEBUG::TimeStreamWriter::writeRecords: batch size [{}], status [{}] ", batchRecords.size(), writeResponse.sdkHttpResponse().statusCode());
-                    } else {
-                        LOGGER.debug("DEBUG::TimeStreamWriter::writeRecords: Batch ingestion is complete for the records of size [{}] ", records.size());
+                    for (Record record : batchRecords) {
+                        RejectedRecord rejectedRecord = new RejectedRecord(record,TimestreamSinkErrorCodes.NO_INGESTION_TARGET);
+                        rejectedRecords.add(rejectedRecord);
                     }
-                } catch (RejectedRecordsException e) {
-                    LOGGER.error("ERROR::TimeStreamWriter::writeRecords: Few records have been rejected in the batch [{}] , due to [{}]", currentBatch, e.getLocalizedMessage());
-                    if (e.hasRejectedRecords()) {
-                        rejectedRecords.addAll(getRejectedTimestreamRecords(e.rejectedRecords(), batchRecords));
+                    LOGGER.error("ERROR::TimeStreamWriter::writeRecords: Records have been rejected in the batch [{}] , due to [{}]", currentBatch, TimestreamSinkErrorCodes.NO_INGESTION_TARGET);
+                }
+                else {
+                    try {
+                        batchRecords = getBatchRecords(records, currentBatch);
```

Comment:
why not generate influxdb write records directly from 'records'

Reply:
Was a possible thought, but wanted to keep logic of max records simple as already done before, otherwise another logic needs to apply of how to limit total number of records to InfluxDB. Certainly open to revisit